htdocs/core/tpl/ajaxrow.tpl.php

```diff
@@ -80,7 +80,12 @@ function() {
 						console.log("tableDND end of ajax call");
 						if (reloadpage == 1) {
 							//console.log('<?php echo $urltorefreshaftermove.' - '.$_SERVER['PHP_SELF'].' - '.dol_escape_js($_SERVER['QUERY_STRING']); ?>');
-							location.href = '<?php echo dol_escape_js(empty($urltorefreshaftermove) ? ($_SERVER['PHP_SELF'].'?'.dol_escape_js($_SERVER['QUERY_STRING'])) : $urltorefreshaftermove); ?>';
+							<?php
+							$redirectURL = empty($urltorefreshaftermove) ? ($_SERVER['PHP_SELF'].'?'.dol_escape_js($_SERVER['QUERY_STRING'])) : $urltorefreshaftermove;
+							// remove action parameter from URL
+							$redirectURL = preg_replace('/[&?]action=[^&]*/', '', $redirectURL);
```

Comment:
Can't you replace
preg_replace('/[&?]action=[^&]*/', '', $redirectURL);
with
preg_replace('/&?action=[^&]*/', '', $redirectURL);
 Why the bracket around first & ?

Reply:
I want "action" to be delimited by either `&` or `?` on the left and I need to take up the whole parameter on the right (delimited by either `&` or by the end of string).

If you remove the square brackets on the left, `page.php?action=abcd&reaction=efgh` would be replaced with `page.php?re`.

My regexp is also incorrect because it removes the leading `?` when "action" is the first parameter (in my example, we would have `page.php&reaction=efgh`).

So to fix this, I have two possibilities:
* `/&?\baction=[^&]*/'` ⇒ valid result (`page.php?&reaction=efgh`) but `?&` is not pretty
* `(?<=\?)action=[^&]*&?|&action=[^&]*` ⇒ slightly better, but less readable

Which one should I use?

Reply:
I suggest this:
preg_replace('/(&|\?)action=[^&]*/', '\1', $redirectURL);
The ? should be escape with \, i think ?

Never mind for page.php?&action=abc    url is valid even if not pretty.

Reply:
I changed it accordingly and also excluded `#` from the matched character set for parameter values, so that if the URL ends with the action parameter followed by a fragment, the fragment will not be removed.